resources/channel.sub.js

```diff
@@ -0,0 +1,1003 @@
+                        result = local;
+                    } else {
+                        result = remote;
+                    }
+                    break;
+                case "sendchannel":
+                    result = new SendChannel(value);
+                    break;
+                case "regexp":
+                    result = new RegExp(value.pattern, value.flags);
+                    break;
+                case "date":
+                    result = new Date(value);
+                    break;
+                case "error":
+                    if (item.value.type in self &&
+                        typeof self[item.value.type] === "function") {
+                        result = new self[item.value.type](item.value.message);
+                    } else {
+                        result = new Error(item.value.message);
+                    }
+                    result.name = item.value.name;
+                    result.lineNumber = item.value.lineNumber;
+                    result.columnNumber = item.value.columnNumber;
+                    result.fileName = item.value.fileName;
+                    result.stack = item.value.stack;
+                    break;
+                case "array":
+                    result = [];
+                    newTarget = {type: "array", value: result};
+                    for (let child of value) {
+                        stack.push({item: child, target: newTarget});
+                    }
+                    break;
+                case "set":
+                    result = new Set();
+                    newTarget = {type: "set", value: result};
+                    for (let child of value) {
+                        stack.push({item: child, target: newTarget});
+                    }
+                    break;
+                case "object":
+                    result = {};
+                    newTarget = {type: "object", value: result};
+                    for (let [targetKey, child] of Object.entries(value)) {
+                        stack.push({item: child, target: newTarget, targetKey});
+                    }
+                    break;
+                case "map":
+                    result = new Map();
+                    newTarget = {type: "map", value: result};
+                    for (let [key, child] of value) {
+                        stack.push({item: key, target: newTarget, targetKey: "key"});
+                        stack.push({item: child, target: newTarget, targetKey: "value"});
+                    }
+                    break;
+                default:
+                    throw new TypeError(`Can't deserialize object of type ${type}`);
+                }
+                if (objectId !== undefined) {
+                    objectMap.set(objectId, result);
+                }
+            }
+
+            if (target === null) {
+                if (deserialized !== null) {
+                    throw new Error("Tried to create multiple output values");
```

Comment:
This is the same error as above, is it accurate here though? What's happened is we we've deserialized a value but there's nowhere to put it.